source/dub/init.d

```diff
@@ -90,6 +90,15 @@ void initPackage(Path root_path, string[string] deps, string type,
 private void initMinimalPackage(Path root_path, ref PackageRecipe p)
 {
 	p.description = "A minimal D application.";
+
+	// we want to create a empty, not-null dictionary,
+	// s. t. the generated recipe contains an empty "dependencies" section
+	// in case of PackageFormat.json
+	auto bar = Dependency("~>0.0.0");
+	p.buildSettings.dependencies["foo"] = bar;
+	p.buildSettings.dependencies.remove("foo");
```

Comment:
On first sight it feels like a too leaky design to me. The approach here is of course also rather hackish, but it has a very specific target. Always outputting a dependencies field on the other hand would affect 90% of cases where this isn't actually needed/desired.